### PR TITLE
Fix #233 #234 move runtime methods into Magnolia.util object

### LINKED ISSUE #233: Magnolia pulls in `scala-reflect` as a runtime classpath dependency
Despite the fact that `scala-reflect` dependency is declared "provided", it's pulled in at runtime:

To confirm:

```
[~] % coursier resolve com.propensive:magnolia_2.13:0.14.5
com.propensive:magnolia_2.13:0.14.5:default
com.propensive:mercator_2.13:0.2.1:default
jline:jline:2.14.6:default
org.scala-lang:scala-compiler:2.13.0:default
org.scala-lang:scala-library:2.13.1:default
org.scala-lang:scala-reflect:2.13.0:default
```

A different library with scala-reflect%Provided does not pull `scala-reflect`:

```
[~] % coursier resolve io.7mind.izumi:fundamentals-reflection_2.13:0.10.3-M2
io.7mind.izumi:fundamentals-collections_2.13:0.10.3-M2:default
io.7mind.izumi:fundamentals-functional_2.13:0.10.3-M2:default
io.7mind.izumi:fundamentals-platform_2.13:0.10.3-M2:default
io.7mind.izumi:fundamentals-reflection_2.13:0.10.3-M2:default
io.7mind.izumi:fundamentals-thirdparty-boopickle-shaded_2.13:0.10.3-M2:default
org.scala-lang:scala-library:2.13.1:default
org.scala-lang.modules:scala-collection-compat_2.13:2.1.4:default
```

# Cause

This happens because `mercator`, which is a dependency of `magnolia`, pulls in scala-reflect at runtime (not as `Provided`):

```
[~] % coursier resolve com.propensive:mercator_2.13:0.2.1
com.propensive:mercator_2.13:0.2.1:default
jline:jline:2.14.6:default
org.scala-lang:scala-compiler:2.13.0:default
org.scala-lang:scala-library:2.13.0:default
org.scala-lang:scala-reflect:2.13.0:default
```

/cc @propensive

### PR DESCRIPTION
**Pull request by neko-kai**
* exclude scala-reflect from mercator dependency

### CODE REVIEW

**Comment by neko-kai** [comment]
Confirmed with manual testing that codecs built against older magnolia correctly call the `private[Magnolia]` methods...

**Comment by joroKr21** [comment]
Thanks 👍 

Looks like we need `scala-reflect` as a dependency in the tests.
That is to be able to test compile time errors.

Also I would vote for moving the helper methods to another file.
Or better yet, why not to `interface`?
The factory methods belong to their respective companion objects IMO.
They have to be public either way so we can't avoid a determined user calling them manually.

**Comment by neko-kai** [comment]
@joroKr21 Moved to `interface.scala/MagnoliaUtil`

**Comment by neko-kai** [comment]
@joroKr21 
> The factory methods belong to their respective companion objects IMO.

Do you want to move them to companion object in this PR or subsequent?

**Comment by joroKr21** [comment]
> Do you want to move them to companion objects in this PR or subsequent?

If we want to move them then this PR makes sense.
I don't feel too strongly about it, just seems more natural.

**Comment by neko-kai** [comment]
Moved

File: build.sbt [hunk 1]
```diff
@@ -9,7 +9,9 @@ lazy val core = crossProject(JVMPlatform, JSPlatform)
   .settings(publishSettings)
   .settings(scalaMacroDependencies)
   .settings(moduleName := "magnolia")
-  .settings(libraryDependencies ++= Seq("com.propensive" %% "mercator" % "0.2.1"))
+  .settings(libraryDependencies ++= Seq(
+    ("com.propensive" %% "mercator" % "0.2.1").exclude("org.scala-lang", "scala-reflect")
```
Comment (propensive) [inline build.sbt:13]: I didn't realise you could do this! Just out of interest, does having `scala-reflect` on the classpath cause any issues, or is it just JAR bloat?

**Comment by propensive** [comment]
Looks good to me! Thank you, @neko-kai, and I'm very happy to have you as a contributor. :)

**Comment by propensive** [comment]
This is going into version 0.15.0, shortly.

File: build.sbt (hunk 1 above)
Comment (neko-kai) [reply build.sbt:13]: Unexpected bloat as I've been trying to keep scala-reflect out of all library dependencies and it just moved back in after adding magnolia.

**Pull request merged**